Tests/ASTableViewTests.mm

```diff
@@ -772,7 +772,12 @@ - (void)testIssue2252
     }];
   });
   dispatch_resume(timer);
-  [self waitForExpectationsWithTimeout:60 handler:nil];
+  // Even if an exception is thrown, we need to cancel the timer so we don't pollute other test methods
+  @try {
+    [self waitForExpectationsWithTimeout:60 handler:nil];
+  } @finally {
+    dispatch_cancel(timer);
+  }
```

Comment:
When I was debugging this diff, I saw that this method would pollute other tests if an exception was thrown during the updating, so I added this in to ensure that we only fail once.